developer/frameworks/articles/data-engine/04-data-layouts.markdown

```diff
@@ -0,0 +1,161 @@
+# Data Layouts
+
+Data Layouts control the display of a from backed by the data engine. The basic
+functionality is to lay the field out in _rows_ and _columns_.
+
+## Adding Data Layouts
+
+Add data layouts by making a POST request to the `data-layouts` endpoint,
+passing in the ID of the data definition the data layout is to be created for:
+
+```sh
+curl -i -X POST \
+    http://localhost:8080/o/data-engine/v1.0/data-definitions/{data_definition_id}/data-layouts \
+    -H 'Authorization: Basic dGVzdEBsaWZlcmF5LmNvbTp0ZXN0' \
+    -H 'Content-Type: application/json' \ 
+{
+    "dataLayoutPages": [
+        {
+            "dataLayoutRows": [
+                {
+                    "dataLayoutColums": [
+                        {
+                            "columnSize": "1",
+                            "fieldNames": [
+                                "fieldName"
+                            ]
+                        }
+                    ]
+                }
+            ],
+            "description": {"en_US":"description"},
+            "title": {"en_US":"title"}
+        }
+    ],
+    "description": {"en_US":"description"},
+    "name": {"en_US":"name"},
+    "paginationMode": "single-page"
+}
+
+```
+
+Here you can create multiple pages, laying out your fields in column and rows.
+Each page can have a title and description, and the `dataLayout` itself can have
+a name and description (a name is required).
+
+
+## Retrieving Data Layouts
+
+### Retrieving all the Data Layouts for a Data Definition
+
+Send a GET request to the `data-definitions` resource's `data-layouts` endpoint,
+passing in the data definition ID as a URL parameter: 
+
+```sh
+curl -i -X GET \
+    http://localhost:8080/o/data-engine/v1.0/data-definitions/{data_definition_id}/data-layouts \
+    -H 'Authorization: Basic dGVzdEBsaWZlcmF5LmNvbTp0ZXN0' \
+    -H 'Content-Type: application/json' \ 
+```
+
+### Retrieving all the Data Layouts in a Site 
+
+Send a GET request to the `data-definitions` resource's `data-layouts` endpoint,
+passing in the data definition ID as a URL parameter: 
+
+```sh
+curl -i -X GET \
+    http://localhost:8080/o/data-engine/v1.0/data-definitions/{data_definition_id}/data-layouts \
```

Comment:
The endpoint is the same as the above (Retrieving all the Data Layouts for a Data Definition).

Should be: sites/{site_id}/data-layout